remix.init/index.js

```diff
@@ -97,83 +84,97 @@ async function installAdditionalDependencies({
 }) {
   try {
     console.log(`Installing additional dependencies with ${packageManager}.`);
-    const npmInstall = await execa(packageManager, ["install"], {
+    const result = await execa(packageManager, ["install"], {
       cwd: rootDirectory,
-      stdio: "inherit",
+      stdio: "ignore",
     });
+    if (result.failed) {
+      return false;
+    }
   } catch (e) {
-    console.log(
-      `Unable to install additional packages. Run ${packageManager} install in the root of the new project, "${rootDirectory}".`
-    );
+    return false;
   }
+  // return true;
+}
+
+async function shouldUseEdge(rootDirectory) {
+  const { "netlify-edge": netlifyEdge } = argh.argv;
+
+  let useEdge = netlifyEdge;
+
+  if (useEdge === undefined) {
+    const projectType = await select({
+      message: "Run your Remix site with:",
+      options: [
+        {
+          value: "functions",
+          label: "Netlify Functions",
+          hint: "A Node-based runtime, running in one region",
+        },
+        {
+          value: "edge",
+          label: "Netlify Edge Functions",
+          hint: "A web-based runtime, running at the edge",
+        },
+      ],
+    });
+    if (isCancel(projectType)) {
+      cancel(
+        `Project setup cancelled. Run remix init inside "${relative(
+          process.cwd(),
+          rootDirectory
+        )}" to complete setup.`
+      );
+      process.exit(1);
+    }
+    useEdge = projectType === "edge";
+  }
+  return useEdge;
 }
 
 async function main({ rootDirectory, packageManager }) {
+  intro(`Welcome to Remix on Netlify`);
+
+  const useEdge = await shouldUseEdge(rootDirectory);
+  const spin = spinner();
+  spin.start("Setting up your project");
+  if (useEdge) {
+    await copyTemplateFiles(rootDirectory);
+  }
```

Comment:
somewhat NIT, but this setup with edge overwriting "default" lambda/function is a bit hard to track (what's different between them, what's common etc)

Reply:
Yeah, I did consider handling all lambda-specific files in the same way as edge, but in the end I wanted to ensure the template would at least build if init hadn't run. An alternative might be to set the `"build"` command to "remix init", and have that overwrite the command. In fact, I think I'll do that.